### PR TITLE
Kube client

### PR DESCRIPTION
**Pull request by aryszka**
as part of #227 and https://github.com/zalando-incubator/kubernetes-on-aws/issues/169

### CODE REVIEW

File: dataclients/kubernetes/kube.go [hunk 1]
```diff
@@ -73,8 +97,13 @@ func New(o Options) *Client {
 		o.KubernetesURL = defaultKubernetesURL
 	}
 
-	log.Debugf("kube client initialized with api address: %s", o.KubernetesURL)
-	return &Client{apiURL: o.KubernetesURL}
+	log.Debugf("kube client initialized with api address: %s; with healtcheck: %t",
```
Comment (lmineiro) [inline dataclients/kubernetes/kube.go]: typo `healtcheck`

File: dataclients/kubernetes/kube.go [hunk 2]
```diff
@@ -279,14 +317,49 @@ func (c *Client) loadAndConvert() ([]*eskip.Route, error) {
 	return r, nil
 }
 
+func healthcheckRoute(healthy bool) *eskip.Route {
+	status := http.StatusOK
+	if !healthy {
+		status = http.StatusNotFound
```
Comment (lmineiro) [inline dataclients/kubernetes/kube.go]: I'm a bit unsure if it shouldn't be a 5XX family error code. Like `503 - Service Unavailable`.
4XX are client errors and this is not the case. In practical terms it serves the same purpose.
What do you think?

File: dataclients/kubernetes/kube_test.go [hunk 3]
```diff
@@ -207,6 +288,12 @@ func (api *testAPI) getTestService(uri string) (*service, bool) {
 }
 
 func (api *testAPI) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if api.failNext {
+		api.failNext = false
+		w.WriteHeader(http.StatusServiceUnavailable)
```
Comment (lmineiro) [inline dataclients/kubernetes/kube_test.go]: http.StatusServiceUnavailable - nice! See my comment above.

File: dataclients/kubernetes/kube.go (hunk 2 above)
Comment (hjacobs) [inline dataclients/kubernetes/kube.go]: Yes, 5xx is more appropriate! (does not make a practical difference, but we should follow HTTP standards..)

File: dataclients/kubernetes/kube.go (hunk 1 above)
Comment (aryszka) [inline dataclients/kubernetes/kube.go]: right

File: dataclients/kubernetes/kube.go (hunk 2 above)
Comment (aryszka) [inline dataclients/kubernetes/kube.go]: ok, thx

**Comment by hjacobs** [comment]
:+1:

**Comment by hjacobs** [comment]
Tests failing..

**Pull request merged**